internal/db/schema/migrations/postgres/dev/80_domains.up.sql

```diff
@@ -0,0 +1,36 @@
+begin;
+
+-- wt_email defines a type for email which must be less than 320 chars and only
+-- contain lower case values.  The type is defined to allow nulls and not be
+-- unique, which can be overriden as needed when used in tables.
+create domain wt_email as text
+    check (length(trim(value)) < 320)
```

Comment:
I think we should also have a `check (length(trim(value)) > 0)` to protect against empty strings.